google-cloud-storage/src/main/java/com/google/cloud/storage/Bucket.java

```diff
@@ -1083,6 +1089,28 @@ public Blob get(String blob, BlobGetOption... options) {
     return storage.get(BlobId.of(getName(), blob), options);
   }
 
+  /**
+   * Returns the requested blob in this bucket of a specific generation or {@code null} if not
+   * found.
+   *
+   * <p>Example of getting a blob of a specific in the bucket.
+   *
+   * <pre>{@code
+   * String blobName = "my_blob_name";
+   * long generation = 42;
+   * Blob blob = bucket.get(blobName, generation);
+   * }</pre>
+   *
+   * @param blob name of the requested blob
+   * @param generation the generation to get
+   * @param options blob search options
+   * @throws StorageException upon failure
+   */
+  @TransportCompatibility({Transport.HTTP, Transport.GRPC})
+  public Blob get(String blob, Long generation, BlobGetOption... options) {
```

Comment:
IIUC this is needed because you can't specify generation on Bucket.get() that exists today; Would recommend using BlobId instead of adding Long generation overload.

Do you think this is hard requirement to support this feature since there's [`storage.get(BlobId..)`](https://github.com/googleapis/java-storage/blob/main/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java#L3033)?

Reply:
I think it would feel pretty weird that soft deleted would become the only state an object can be in where you can get it from `Storage` but can't get it from the `Bucket` that the object lives in.

I also think using `BlobId` here would feel weird, because you're already calling from a Bucket, so it's weird to have to supply the same Bucket name into the BlobId. There are no other methods in `Bucket.java` that have a `BlobId` in the signature

Reply:
Bleh, missed that bucket name is also part of BlobId. thanks.
We haven't had generation in Bucket.get() at all which is required for version enabled buckets, that's why i asked this question as well.

For the sake of discussion, WDYT about having generation under BlobGetOption's? It's a query paramter just like IfGenerationMatch etc.